mantidproject/mantid
# Combine OS X and Linux compiler flags.

Description of work.

This simplifies the CMake script by setting flags for clang and gcc on OS X and Linux in a single location. I suppressed the remaining `-Wcast-align` warnings from `Framework/ICat/src/GSoap/stdsoap2.cpp` 

I'm also sneaking in some fixes for `-Winconsistent-missing-override` warnings that are now in master.

**To test:**


- Check that the OS X compiler flags are still being set (I suggest using the option `CMAKE_EXPORT_COMPILER_COMMANDS` and viewing the json file)
- Verify that suppressing `-Wcast-align` warnings from `Framework/ICat/src/GSoap/stdsoap2.cpp` is acceptable.
- Check that Xcode still picks up the correct flags, the C++14 standard and libc++ standard library.

Fixes #11681.

Does not need to be in the release notes.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
